tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -1002,6 +1002,159 @@ TYPED_TEST(
+{
+	const auto m = 2;
+	SGMatrix<TypeParam> A(m, m);
+	SGMatrix<TypeParam> B(m, m);
+	SGMatrix<TypeParam> result(m, m);
+
+	for (auto i : range(m * m))
+	{
+		A[i] = i;
+		B[i] = i + 1;
+	}
+	
+	element_div(A, B, result);
+	for (auto i : range(m))
+		for (auto j : range(m))
+			EXPECT_NEAR(
+			    result(i, j), A(i, j) / B(i, j), get_epsilon<TypeParam>());
+	
+	SGMatrix<TypeParam> C(m+1, m);
+	SGMatrix<TypeParam> D(m, m+1);
+	SGMatrix<TypeParam> E(m+1, m+1);
+
+	EXPECT_THROW(element_div(A, B, C), ShogunException);
+	EXPECT_THROW(element_div(A, B, D), ShogunException);
+	EXPECT_THROW(element_div(A, B, E), ShogunException);
+
+	EXPECT_THROW(element_div(A, C, B), ShogunException);
+	EXPECT_THROW(element_div(A, D, B), ShogunException);
+	EXPECT_THROW(element_div(A, E, B), ShogunException);
+
+	EXPECT_THROW(element_div(C, A, B), ShogunException);
+	EXPECT_THROW(element_div(D, A, B), ShogunException);
+	EXPECT_THROW(element_div(E, A, B), ShogunException);
+}
+
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGMatrix_block_elementwise_division)
+{
+	const index_t nrows = 2;
+	const index_t ncols = 3;
+
+	SGMatrix<TypeParam> A(nrows, ncols);
+	SGMatrix<TypeParam> B(ncols, nrows);
+
+	for (auto i : range(nrows))
+		for (auto j : range(ncols))
+		{
+			A(i, j) = i * 10 + j + 1;
+			B(j, i) = i + j + 1;
+		}
+
+	const auto m = 2;
+	auto A_block = linalg::block(A, 0, 0, m, m);
+	auto B_block = linalg::block(B, 0, 0, m, m);
+	auto result = element_div(A_block, B_block);
+
+	ASSERT_EQ(result.num_rows, m);
+	ASSERT_EQ(result.num_cols, m);
+
+	for (auto i : range(m))
+		for (auto j : range(m))
+			EXPECT_NEAR(
+			    result(i, j), A(i, j) / B(i, j), get_epsilon<TypeParam>());
+
+	auto C_block = linalg::block(B, 0, 0, m, m+1);
+	EXPECT_THROW(element_div(A_block, C_block),ShogunException);
+	C_block = linalg::block(B, 0, 0, m+1, m);
+	EXPECT_THROW(element_div(A_block, C_block),ShogunException);
+	C_block = linalg::block(B, m-1, 0, m, m+1);
+	EXPECT_THROW(element_div(A_block, C_block),ShogunException);
+
+}
+
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGVector_elementwise_division)
+{
+	const index_t len = 4;
+	SGVector<TypeParam> a(len);
+	SGVector<TypeParam> b(len);
+	SGVector<TypeParam> c(len);
+
+	for (index_t i = 0; i < len; ++i)
+	{
+		a[i] = i;
+		b[i] = i + 1;
+	}
+
+	c = element_div(a, b);
+
+	for (index_t i = 0; i < len; ++i)
+		EXPECT_NEAR(a[i] / b[i], c[i], get_epsilon<TypeParam>());
+
+	SGVector<TypeParam> d(len-1);
+	EXPECT_THROW(element_div(a, d), ShogunException);
+	EXPECT_THROW(element_div(d, a), ShogunException);
+}
+
+TYPED_TEST(
+    LinalgBackendEigenAllTypesTest, SGVector_elementwise_division_in_place)
+{
+	const index_t len = 4;
+	SGVector<TypeParam> a(len);
+	SGVector<TypeParam> b(len);
+	SGVector<TypeParam> c(len);
+
+	for (index_t i = 0; i < len; ++i)
+	{
+		a[i] = i;
+		b[i] = i + 1;
+		c[i] = i;
+	}
+
+	element_div(a, b, a);
+	for (index_t i = 0; i < len; ++i)
+		EXPECT_NEAR(c[i] / b[i], a[i], get_epsilon<TypeParam>());
```

Comment:
btw, there is a `zip_iterator`, so you can write this python style
```cpp
for (const auto& [result, el1, el2]: zip_iterator(a,b,c))
    EXPECT_NEAR(el2[i] / el1y[i], result[i], get_epsilon<TypeParam>());
```
Same in all the other loops where you check the result. Another thing that you should try to use in your code, because it cleans it up a lot